plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -334,12 +341,20 @@ DO UPDATE SET property_type=$5, property_type_format=$6 WHERE posthog_propertyde
                 'fetchPropertyDefinitions'
             )
 
-            this.propertyDefinitionsCache.set(
-                teamId,
-                eventProperties.rows.reduce((teamCache, row) => {
-                    teamCache[row.name] = row.property_type ?? NULL_IN_DATABASE
-                    return teamCache
-                }, new Map())
+            const teamPropertyDefinitionsCache = new LRU<string, string | symbol>({
+                max: this.lruCacheSize * 10, // keep in memory the last 100k property definitions we have seen
+                maxAge: ONE_HOUR * 24, // cache up to 24h
+                updateAgeOnGet: true,
+            })
+
+            eventProperties.rows.forEach((row) => {
```

Comment:
I usually like simple `for of` loops, unless I'm in some functional environment. There's still [some overhead](https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/#arrayforeach-vs-for-and-forof) in creating and calling a function, even if V8 is pretty good in inlining things. `for of` loops give you a much nicer syntax for looping over object entries as well... but they don't come with `index`.

On the frontend it doesn't really matter, but here I think tiny quick wins make sense.

Reply:
> I usually like simple for of loops

I'll fight you to the death...

Hmmm, well, I suppose I only have to press ALT and ENTER to switch between them :)